routes/userStatus.js

```diff
@@ -15,7 +16,13 @@ const { validateUserStatus } = require("../middlewares/validators/userStatus");
 router.get("/", getAllUserStatus);
 router.get("/self", authenticate, getUserStatus);
 router.get("/:userId", getUserStatus);
-router.patch("/self", authenticate, validateUserStatus, updateUserStatus);
+router.patch("/self", authenticate, validateUserStatus, (req, res, next) => {
+  if (Object.keys(req.body).includes("cancelOoo")) {
+    cancelOOOStatus(req, res, next);
+  } else {
+    updateUserStatus(req, res, next);
+  }
+});
```

Comment:
I believe this should be moved to the controller. Or some middleware instead of adding it to routes. I could be wrong.

Reply:
I don't think routes should have these logic that handles this.

Reply:
Makes sense. Moved them to the controller layer. Please re-review.